FullStackMVCAPP/FullStackMVCAPP/Repositories/Base/IEntityRepository.cs

```diff
@@ -0,0 +1,12 @@
+﻿using FullStackMVCAPP.DataContext;
+using System;
+using System.Collections.Generic;
+
+namespace FullStackMVCAPP.Repositories.Base
+{
+public interface IEntityRepository<T> where T : class
+{
+    IList<T> EntityList(GOTContext gameOfThronesContext);
+    T GetEntityByID(GOTContext gameOfThronesContext,int id);
```

Comment:
Interface implements a generic (`T`) however, the repository itself returns a specific `Castle` entities. Would be best to remove the generics and replace with exactly what it should be returning to avoid confusion.

Reply:
But House and Character enherit both from this, wouldn't that cause a problem?

Reply:
Look more at what is returned by the method. Even if the method itself is doing a few different things, its whatever is returned that is key here.